File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutBundle.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;

public class PutBundle extends PutBase<List<PutBase<?>>> {

  public static final String JAVA_CLASS = "PutBundle";

  public PutBundle(String key, List<PutBase<?>> value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutBundle putBundle) {
    Bundle subBundle = new Bundle();
    for (PutBase<?> putBase : putBundle.value) {
      PutBase.addToAndroidOsBundle(subBundle, putBase);
    }
    bundle.putBundle(putBundle.key, subBundle);
  }

  public static PutBundle fromJson(JSONObject jsonObject) throws JSONException {
    final ArrayList<PutBase<?>> arrayList = new ArrayList<>();
    final JSONArray jsonArray = jsonObject.getJSONArray(Constants.VALUE);
    for (int i = 0; i < jsonArray.length(); i++) {
      arrayList.add(PutBase.fromJson(jsonArray.getJSONObject(i)));
    }
    return new PutBundle(jsonObject.getString(Constants.KEY), arrayList);
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    final JSONArray jsonArray = new JSONArray();
    for (PutBase<?> putBase : value) {
      jsonArray.put(putBase.toJson());
    }
    jsonObject.put("value", jsonArray);
    return jsonObject;
  }
}

File: packages/android_intent_plus/lib/src/put_classes/put_bundle.dart
import 'package:android_intent_plus/android_intent.dart';
import 'package:android_intent_plus/src/put_classes/base/put_base.dart';

class PutBundle extends PutBase<List<PutBase>> {
  PutBundle({required String key, required List<PutBase> value})
      : super(key: key, value: value);

  @override
  String get javaClass => 'PutBundle';

  factory PutBundle.fromJson({required String key, required dynamic value}) {
    return PutBundle(key: key, value: Bundle.fromJson(value).value);
  }

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'key': key,
      'javaClass': javaClass,
      'value': value.map((e) => e.toJson()).toList(),
    };
  }
}
